src/Angor/Shared/WalletOperations.cs

```diff
@@ -128,10 +127,92 @@ public string GenerateWalletWords()
             .SendEstimatedFees(new FeeRate(Money.Coins(sendInfo.FeeRate)));
 
         var signedTransaction = builder.BuildTransaction(true);
-        
+
         return await PublishTransactionAsync(network, signedTransaction);
     }
 
+    private void UpdateAccountPendingLists(AccountInfo accountInfo, UnconfirmedInfo unconfirmedInfo)
+    {
+        // remove from the pending remove list if it was removed from the indexer
+        var pendingRemove = unconfirmedInfo.AccountPendingSpent.ToList();
+        foreach (var utxoData in pendingRemove)
+        {
+            foreach (var addressInfo in accountInfo.AddressesInfo.Concat(accountInfo.ChangeAddressesInfo))
+            {
+                if (addressInfo.Address == utxoData.address)
+                {
+                    if (addressInfo.UtxoData.All(_ => _.outpoint.ToString() != utxoData.outpoint.ToString()))
+                    {
+                        unconfirmedInfo.AccountPendingSpent.Remove(utxoData);
+                    }
+                }
+            }
+        }
+
+        // remove from the pending add if it was removed from the indexer
+        var pendingAdd = unconfirmedInfo.AccountPendingReceive.ToList();
+        foreach (var utxoData in pendingAdd)
+        {
+            foreach (var addressInfo in accountInfo.AddressesInfo.Concat(accountInfo.ChangeAddressesInfo))
+            {
+                if (addressInfo.Address == utxoData.address)
+                {
+                    if (addressInfo.UtxoData.Any(_ => _.outpoint.ToString() == utxoData.outpoint.ToString()))
+                    {
+                        unconfirmedInfo.AccountPendingReceive.Remove(utxoData);
+                    }
+                }
+            }
+        }
+    }
+
+    public void UpdateAccountUnconfirmedInfoWithSpentTransaction(AccountInfo accountInfo, UnconfirmedInfo unconfirmedInfo, Transaction transaction)
+    {
+        Network network = _networkConfiguration.GetNetwork();
+
+        var outputs = transaction.Outputs.AsIndexedOutputs();
+        var inputs = transaction.Inputs.Select(_ => _.PrevOut).ToList();
+        
+        foreach (var addressInfo in accountInfo.AddressesInfo.Concat(accountInfo.ChangeAddressesInfo))
+        {
+            // find all spent inputs to mark them as spent
+            foreach (var utxoData in addressInfo.UtxoData)
+            {
+                foreach (var outPoint in inputs)
```

Comment:
3 nested loops, isn't there a better way to get the utxos?

Reply:
Even if I use linq there will still be loops it will just look shorter.
To optimize here we will need to use dictionaries but I think this is not really needed